### ISSUE TITLE
Why extent 4096?

### ISSUE THREAD

**Issue by rouen-sk** [mapbox/vector-tiles#45]
Hi, this is not an issue, just a question. What is the reasoning behind the value 4096 as default extent? I have worked with raster tiles maps and now looking into vector maps, and this is not clear to me. As I understand the MVT format, real-world coordinates are projected into arbitrary integer coordinates, and this is nice because it "simplifies" geometries. Now, super basic would be extent "corresponding" to pixels: 256. Since Mapbox GL JS zoom is different from slippy maps standard (mapbox zoom 13 is the same as slippy maps 14), I would understand extent 512.. so, why 4096? (which is like 8 "subpixels" accuracy)

**Comment by GretaCB** [comment]
Hey @rouen-sk , great question. `4096` comes from 256 * 16. This means: 
- You can go 4 zoom levels of [overzooming](https://github.com/mapbox/vector-tiles/issues/21) without changing what pixel it belongs to in the final image. 
- So if you had data at zoom level 18, you could go to z22 overzooming that tile with out seeing any difference compared to where the same pixel would be if we had a tile at zoom 22 at zoom lvl 18
- Depending on where you are, the width of that tile is about ~150 meters, this means that you have 150 / 4096 = 0.0366 ~= 0.04. So you have 4 cm precision at zoom level 18 with 4096 extent
- Obviously most maps don’t go to zoom level 18, but you get the idea here — 4096 is precise enough for most data to a certain point

So, to summarize, when testing rendering during the first iteration of the spec, we found that 4096 was a large enough extent to retain precision enough for subjectively high quality visual rendering even when tiles are overzoomed by several zoom levels.
